pkg/api/types.go

```diff
@@ -172,6 +172,9 @@ type Volume struct {
 	// This is optional for now. If not specified, the Volume is implied to be an EmptyDir.
 	// This implied behavior is deprecated and will be removed in a future version.
 	VolumeSource `json:",inline,omitempty"`
+	// SecretName is the name of a Secret in the same namespace as the pod containing this volume
+	// used for volume authentication
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
@pmorie agree re: validation, but I think that's for validation.go (e.g, validateGCEPersistentDiskVolumeSource).